### PR TITLE
feat(sidecar): reconcile WS topic allow-list + drift-guard (#154 Part D step 1)

### PR DESCRIPTION
**Pull request by agorokh**
First step of EPIC #154 **Part D** (make the declared WS topics real), and a standalone correctness fix.

## Problem
`external_protocol.KNOWN_TOPICS` — the allow-list `validate_inbound` consults for `state.subscribe` — listed only the 5 *declared* topics (`connection/session/lap/delta/tire_temps`) and **omitted the two topics the trainer actually publishes**: `coaching.snapshot` (coaching_publisher.lua) and `setup.active` (ac_copilot_trainer.lua). Fan-out (`_broadcast_external`) is topic-agnostic so frames still reached peers, but a client could **never legitimately `state.subscribe`** to them — the "produced-but-unsubscribable" sibling of the #170 handshake bug found in-sim this session.

## Change
- Add `coaching.snapshot` + `setup.active` to `KNOWN_TOPICS` (honest 7-name set). No fan-out behaviour change — `state.subscribe` stays advisory.
- New `tests/test_ws_topic_allowlist.py`:
  - `validate_inbound` accepts a subscribe to every `KNOWN_TOPICS` entry, rejects unknown.
  - **Drift-guard** — statically resolves every `publishTopic` topic in the Lua sources (inline literal *and* `local TOPIC = "..."` constant, e.g. coaching_publisher's) and asserts each ∈ `KNOWN_TOPICS`. Fails CI the moment a producer is wired without its allow-list entry — the structural fix for the recurring "forgot the allow-list" pitfall.

## Verification
Pure-Python / no game: `pytest tests/test_ws_topic_allowlist.py` (10 passed). Prepares Part D's producer wiring (connection/session/lap/delta/tire_temps), tracked in the vault handoff.

Relates to #154.

### CODE REVIEW

File: tests/test_ws_topic_allowlist.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    topics: dict[str, str] = {}
+    unresolved: list[str] = []
+    for f in sorted(LUA_SRC.rglob("*.lua")):
+        text = f.read_text(encoding="utf-8")
+        for m in _CALL_RE.finditer(text):
+            line_start = text.rfind("\n", 0, m.start()) + 1
+            line_end = text.find("\n", m.start())
+            line = text[line_start : line_end if line_end != -1 else len(text)]
+            if "function" in line:  # the definition, not a call
+                continue
```
[the remark this answers is not in the document]
Comment (agorokh) [reply tests/test_ws_topic_allowlist.py]: Good call — fixed in 0efda56. The drift-guard now also skips a publishTopic call when a Lua '--' comment prefix precedes it on the line, so commented-out calls can't trip false positives.

File: tests/test_ws_topic_allowlist.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
```
[the remark this answers is not in the document]
Comment (agorokh) [reply tests/test_ws_topic_allowlist.py]: Fixed in d2d0978. Replaced the broad 'function in line' check with a precise one: exclude a match only when the receiver is immediately preceded by 'function ' (the function M.publishTopic(...) definition), via _DEF_RE. A real .publishTopic( call on a pcall(function() ...) line is now correctly scanned. Verified the guard still detects {coaching.snapshot, setup.active}.

File: tests/test_ws_topic_allowlist.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    unresolved: list[str] = []
+    for f in sorted(LUA_SRC.rglob("*.lua")):
+        text = f.read_text(encoding="utf-8")
+        for m in _CALL_RE.finditer(text):
+            line_start = text.rfind("\n", 0, m.start()) + 1
+            line_end = text.find("\n", m.start())
+            line = text[line_start : line_end if line_end != -1 else len(text)]
+            prefix = line[: m.start() - line_start]  # line text before this `.publishTopic(`
+            if "--" in prefix:  # commented-out call (Lua --)
+                continue
```
[the remark this answers is not in the document]
Comment (agorokh) [reply tests/test_ws_topic_allowlist.py]: Fixed in 04273b2. The line prefix now has string literals stripped (_STR_RE) before the comment/definition checks, so a '--' (or 'function') inside a string no longer false-triggers. Added a self-test (test_drift_guard_resolves_calls_and_skips_noise) pinning all the edge cases: const-resolved topic, inline literal, a call on a pcall(function() ...) line, '--' inside a string, a commented-out call, and the publishTopic definition.

File: tests/test_ws_topic_allowlist.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            continue
+        if _DEF_RE.search(prefix):  # the `function M.publishTopic(...)` definition, not a call
+            continue
+        arg = m.group(1)
+        if arg[:1] in ("'", '"'):
+            topics.add(arg.strip("'\""))
+            continue
+        vm = re.search(rf"\b{re.escape(arg)}\s*=\s*['\"]([^'\"]+)['\"]", text)
+        if vm:
+            topics.add(vm.group(1))
```
[the remark this answers is not in the document]
Comment (agorokh) [reply tests/test_ws_topic_allowlist.py:108]: Fixed in 06736e9. The guard now strips Lua line comments once up front (string-aware, honoring escaped quotes) BEFORE any scanning, so a commented-out '-- TOPIC = ...' decoy assignment is gone before const-resolution runs (and commented calls likewise). The self-test now pins this: a leading '-- TOPIC = "decoy_wrong"' resolves to the real coaching.snapshot, not the decoy. Block comments (--[[ ]]) are documented as out of scope (no producer wraps a publishTopic in one).
